event/src/app/api/events/[id]/route.ts

```diff
@@ -0,0 +1,69 @@
+import { IUserPayload } from "@/app/api/users/login/route";
+import { NextRequest, NextResponse } from "next/server";
+import * as jose from 'jose'
+import Event, { Events } from "@/models/Event";
+import dbConnect from "@/lib/mongo";
+import { EventSchema, IEvent } from "../host/route";
+
+export async function DELETE(req: NextRequest, content: any) {
+  let skey: string = process.env.SECRETKEY!;
+  let cookie = req.cookies.get("accessToken")
+  const key = new TextEncoder().encode(skey)
+  if (!cookie) {
+    console.log('no cookie')
+    return NextResponse.json({
+      status: 400,
+      body: { error: 'Data Invalid', details: 'Not Authorized' },
+    })
+  }
+
+  const { payload, protectedHeader }: { payload: IUserPayload, protectedHeader: any } = await jose.jwtVerify(cookie.value, key, {})
+  if (!payload.userId) {
+    return NextResponse.json({ error: "Invalid user" })
+  }
```

Comment:
Looks fine, but suggest to also check payload.isSubscribed (also consider renaming isSubscribed to isHost since that makes more sense...)